source/gpus_in_openstack.rst

```diff
@@ -4,6 +4,601 @@
+##################
+
+Bios configuration
+------------------
+
+Intel
+^^^^^
+
+* Enable `VT-x` in the BIOS for virtualisation support.
+* Enable `VT-d` in the BIOS for IOMMU support.
+
+Dell
+^^^^
+
+Enabling SR-IOV with `racadm`:
+
+.. code:: shell
+
+    /opt/dell/srvadmin/bin/idracadm7 set BIOS.IntegratedDevices.SriovGlobalEnable Enabled
+    /opt/dell/srvadmin/bin/idracadm7 jobqueue create BIOS.Setup.1-1
+    <reboot>
+
+
+Obtain driver from NVIDIA licensing portal
+-------------------------------------------
+
+Download Nvidia GRID driver from `here <https://docs.nvidia.com/grid/latest/grid-software-quick-start-guide/index.html#redeeming-pak-and-downloading-grid-software>`__
+(This requires a login). The file can either be placed on the :ref:`ansible control host<NVIDIA control host>` or :ref:`uploaded to pulp<NVIDIA Pulp>`.
+
+.. _NVIDIA Pulp:
+
+Uploading the GRID driver to pulp
+---------------------------------
+
+Uploading the driver to pulp will make it possible to run kayobe from any host. This can be useful when
+running in a CI environment.
+
+.. code:: shell
+
+    pulp artifact upload --file ~/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip
+    pulp file content create --relative-path "NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip" --sha256 c8e12c15b881df35e618bdee1f141cbfcc7e112358f0139ceaa95b48e20761e0
+    pulp file repository create --name nvidia
+    pulp file repository add --name nvidia --sha256 c8e12c15b881df35e618bdee1f141cbfcc7e112358f0139ceaa95b48e20761e0 --relative-path "NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+    pulp file publication create --repository nvidia
+    pulp file distribution update --name nvidia --base-path nvidia --repository nvidia
+
+The file will then be available at ``<pulp_url>/pulp/content/nvidia/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip``. You
+will need to set the ``vgpu_driver_url`` configuration option to this value:
+
+.. code:: yaml
+
+   # URL of GRID driver in pulp
+   vgpu_driver_url: "http://{{ pulp_url }}/pulp/content/nvidia/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+
+See :ref:`NVIDIA Role Configuration`.
+
+.. _NVIDIA control host:
+
+Placing the GRID driver on the ansible control host
+---------------------------------------------------
+
+Copy the driver bundle to a known location on the ansible control host. Set the ``vgpu_driver_url`` configuration variable to reference this
+path using ``file`` as the url scheme e.g:
+
+.. code:: yaml
+
+    # Location of NVIDIA GRID driver on localhost
+    vgpu_driver_url: "file://{{ lookup('env', 'HOME') }}/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+
+See :ref:`NVIDIA Role Configuration`.
+
+.. _NVIDIA OS Configuration:
+
+OS Configuration
+----------------
+
+Host OS configuration is done by using roles in the `stackhpc.linux <https://github.com/stackhpc/ansible-collection-linux>`_ ansible collection.
+
+Add the following to your ansible ``requirements.yml``:
```

Comment:
Could these changes go into s-k-c?

Reply:
For me it depends on when we expect to get it into Kayobe. If we don't think we'll have it in time to backport to the (downstream) Antelope release, then I'd suggest we commit the custom playbook. It can always be removed in favour of the Kayobe support in future, and keeps everyone in sync.

FWIW, I didn't put the Kayobe VGPU support in the infra roadmap, thinking we'd probably end up moving on after getting to this point. That said, if you think it's a reasonably small piece of work then it might be worth doing in time to backport to antelope downstream.